docs/custom_loading.rst

```diff
@@ -88,11 +88,59 @@ ensure that the data file being loaded is compatible with the loader function.
             meta = {'header': header}
             wcs = WCS(hdulist[0].header)
             uncertainty = StdDevUncertainty(tab["err"])
-            data = tab["flux"] * Unit("Jy")
+            lamb = Quantity(tab["spectral_axis"])
+            data = Quantity(tab["flux"])
 
-        return Spectrum1D(flux=data, wcs=wcs, uncertainty=uncertainty, meta=meta)
+        return Spectrum1D(flux=data, spectral_axis=lamb, wcs=wcs, uncertainty=uncertainty, meta=meta)
 
 
+The above is based on the ``Quantity`` formulation, as for any reasonably "generic" format
+``Table.read()`` would extract the units from the FITS header anyway.
+
+Another approach is based on the ``Table`` formulation, as follows:
+
+.. code-block:: python
+
+    import os
+
+    from astropy.io import fits
+    from astropy.nddata import StdDevUncertainty
+    from astropy.table import Table
+    from astropy.units import Unit
+    from astropy.wcs import WCS
+
+    from specutils.io.registers import data_loader
+    from specutils import Spectrum1D
+
+
+    # Define an optional identifier. If made specific enough, this circumvents the
+    # need to add ``format="my-format"`` in the ``Spectrum1D.read`` call.
+    def identify_generic_fits(origin, *args, **kwargs):
+        return (isinstance(args[0], str) and
+                os.path.splitext(args[0].lower())[1] == '.fits')
+
+    @data_loader("my-format", identifier=identify_generic_fits,
+                 extensions=['fits'])
+    def generic_fits(file_name, **kwargs):
+        with fits.open(file_name, **kwargs) as hdulist:
+
+            tab = Table.read(file_name)
+
+            uncertainty = StdDevUncertainty(tab["err"])
+            lamb = tab["spectral_axis"] * Unit('')
+            data = tab["flux"] * Unit('')
+
+        return Spectrum1D(flux=data, spectral_axis=lamb, uncertainty=uncertainty)
+
+
+The above example makes the assumption that the units of the wavelength and the flux are
+stored in the table's ``spectral_axis`` and ``flux`` columns as ``Quantity`` objects.
```

Comment:
I'm confused by this statement, as `lamb` and `data` are both being made into `Quantity` objects using explicitly empty `Unit` instances. How are we assuming that the units of the wavelength and flux are in the columns?